src/pal/pal_windows.h

```diff
@@ -20,23 +20,31 @@ namespace snmalloc
 {
   class PALWindows
   {
-    /**
-     * The number of times that the memory pressure notification has fired.
-     */
-    static inline std::atomic<uint64_t> pressure_epoch;
     /**
      * A flag indicating that we have tried to register for low-memory
      * notifications.
      */
     static inline std::atomic<bool> registered_for_notifications;
     static inline HANDLE lowMemoryObject;
+
+    /**
+     * List of callbacks for low-memory notification
+     **/
+    static inline std::atomic<PalNotificationObject*> low_memory_callbacks =
+      nullptr;
+
     /**
      * Callback, used when the system delivers a low-memory notification.  This
-     * simply increments an atomic counter each time the notification is raised.
+     * calls all the handlers registered with the PAL.
      */
     static void CALLBACK low_memory(_In_ PVOID, _In_ BOOLEAN)
     {
-      pressure_epoch++;
+      PalNotificationObject* curr = low_memory_callbacks;
```

Comment:
Can this and the list be pulled out into a separate class?  I'd rather not have to copy and paste this for other platforms.

Reply:
Done